### PR TITLE
ENCD-5898-separate-ami-demo-prod

### PR DESCRIPTION
**Pull request by ottojolanki**

### CODE REVIEW

File: src/encoded/commands/deploy.py [hunk 1]
```diff
@@ -1077,7 +1077,14 @@ def hostname(value):
...
     if not args.image_id:
-        current_ami_id = _fetch_ami_id()
+        if args.profile_name == "default":
+            print("Fetching development AMI.")
+            current_ami_id = _fetch_ami_id()
+        elif args.profile_name == "production":
+            print("Fetching production AMI.")
+            current_ami_id = _fetch_ami_id(key="ami-id/production/current_ami_id.txt")
+        else:
+            raise ValueError("--profile-name must be default or production")
```
Comment (paul-sud) [inline src/encoded/commands/deploy.py]: If it's not hard, maybe we could make this into `choices` in the argument parser?

**Review by paul-sud — CHANGES_REQUESTED** [review]
One potential change requested.

File: src/encoded/commands/deploy.py (hunk 1 above)
Comment (ottojolanki) [reply src/encoded/commands/deploy.py]: Yea, good point. That is the way.

File: src/encoded/commands/deploy.py [hunk 2]
```diff
@@ -1024,7 +1024,7 @@ def hostname(value):
     parser.add_argument('--wale-s3-prefix', default='s3://encoded-backups-prod/production-pg11')
 
     # AWS
-    parser.add_argument('--profile-name', default='default', help="AWS creds profile")
+    parser.add_argument('--profile-name', default='default', choices = ("default", "production"), help="AWS creds profile")
```
Comment (paul-sud) [inline src/encoded/commands/deploy.py]: Formatting nit, there shouldn't be spaces around the `=` for kwargs:
```suggestion
    parser.add_argument('--profile-name', default='default', choices=("default", "production"), help="AWS creds profile")
```

**Review by paul-sud — CHANGES_REQUESTED** [review]
One formatting change requested

**Review by paul-sud — DISMISSED** [review]

File: src/encoded/commands/deploy.py (hunk 2 above)
Comment (ottojolanki) [reply src/encoded/commands/deploy.py]: correct. man, we really need to get some automatic formatting going in this repo.

Comment (paul-sud) [reply src/encoded/commands/deploy.py]: For sure, the current formatting is a mess.

**Review by paul-sud — APPROVED** [review]

**Pull request merged**